Libplanet.Standalone/Hosting/LibplanetNodeService.cs

```diff
@@ -120,80 +122,18 @@ public class LibplanetNodeService<T> : IHostedService, IDisposable
 
         public async Task StartAsync(CancellationToken cancellationToken)
         {
-            var peers = _properties.Peers.ToImmutableArray();
-
-            Task BootstrapSwarmAsync(int depth)
-                => Swarm.BootstrapAsync(
-                    peers,
-                    pingSeedTimeout: PingSeedTimeout,
-                    findNeighborsTimeout: FindNeighborsTimeout,
-                    depth: depth,
-                    cancellationToken: cancellationToken
-                );
-
-            if (peers.Any())
-            {
-                try
-                {
-                    await BootstrapSwarmAsync(1);
-                    BootstrapEnded.Set();
-                }
-                catch (PeerDiscoveryException e)
-                {
-                    Log.Error(e, "Bootstrap failed: {Exception}", e);
-
-                    if (!_ignoreBootstrapFailure)
-                    {
-                        throw;
-                    }
-                }
-
-                await Swarm.PreloadAsync(
-                    TimeSpan.FromSeconds(5),
-                    _preloadProgress,
-                    _properties.TrustedStateValidators,
-                    cancellationToken: cancellationToken
-                );
-                PreloadEnded.Set();
-            }
-            else
-            {
-                BootstrapEnded.Set();
-                PreloadEnded.Set();
-            }
-
-            async Task ReconnectToSeedPeers(CancellationToken token)
-            {
-                while (!token.IsCancellationRequested)
-                {
-                    await Task.Delay(BootstrapInterval);
-                    await BootstrapSwarmAsync(0).ContinueWith(t =>
-                    {
-                        if (t.IsFaulted)
-                        {
-                            Log.Error(t.Exception, "Periodic bootstrap failed.");
-                        }
-                    });
-                    token.ThrowIfCancellationRequested();
-                }
-            }
-            _swarmCancellationToken = cancellationToken;
-            try
+            bool preload = true;
+            while (!cancellationToken.IsCancellationRequested && !_stopRequested)
             {
                 await await Task.WhenAny(
-                    Swarm.StartAsync(
-                        cancellationToken: cancellationToken,
-                        millisecondsBroadcastTxInterval: 15000
-                    ),
-                    ReconnectToSeedPeers(cancellationToken)
+                    StartSwarm(preload, cancellationToken),
+                    CheckSwarm(cancellationToken)
```

Comment:
What if `CheckSwarm()` throws during IBD?

Reply:
In IBD, [this check](https://github.com/planetarium/NineChronicles.Standalone/pull/24/files#diff-b2f7c092a968f1c50ebb861d81eaa797R288) will fail(`false`) because `Swarm.LastMessageTimestamp` will be `null`.

but, do you think it would be better if we have an explicit guard for IBD?

Reply:
I guess this is enough because `Swarm.LastMessageTimestamp` is only renewed when non-reply messages are received. During IBD, only reply messages are needed.